library/Jira/Web/RenderingHelper.php

```diff
@@ -17,49 +19,207 @@ class RenderingHelper
+                || strpos($match[2], 'icingaweb2/monitoring/service/show') !== false;
+
+            $icingadbObjectLink = strpos($match[2], 'icingaweb2/icingadb/host') !== false
+                || strpos($match[2], 'icingaweb2/icingadb/service') !== false;
+
+            if ($monitoringObjectLink || $icingadbObjectLink) {
+                if (strpos($match[2], 'service') !== false) {
+                    if ($monitoringObjectLink) {
+                        $urlServiceParam = $url->getParam('service');
+                        $urlHostParam = $url->getParam('host');
+                        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
+                            $link->setUrl(Url::fromPath(
+                                'icingadb/service',
+                                [
+                                    'name'      => $urlServiceParam,
+                                    'host.name' => $urlHostParam,
+                                ]
+                            ));
+                        }
+                    } else {
+                        $urlServiceParam = $url->getParam('name');
+                        $urlHostParam = $url->getParam('host.name');
+                    }
+
+                    if ($urlServiceParam === $this->serviceName && $urlHostParam === $this->hostName) {
+                        $serviceLink = clone $link;
+                        $serviceLink->setContent([new Icon('cog'), $match[1]])
+                            ->addAttributes(['title' => t('Show Icinga Service State')]);
+                        $this->setServiceLink($serviceLink);
+                    }
+                } else {
+                    $icon = new Icon('server');
+                    if ($monitoringObjectLink) {
+                        $urlHostParam = $url->getParam('host');
+                        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
+                            $link->setUrl(Url::fromPath(
+                                'icingadb/host',
+                                ['name' => $urlHostParam]
+                            ));
+                        } else {
+                            $icon = new Icon('laptop');
+                        }
+                    } else {
+                        $urlHostParam = $url->getParam('name');
+                    }
+
+                    if ($urlHostParam === $this->hostName) {
+                        $hostLink = clone $link;
+                        $hostLink->setContent([$icon, $match[1]])
+                            ->addAttributes(['title' => t('Show Icinga Host State')]);
+                        $this->setHostLink($hostLink);
+                    }
+                }
+            } elseif ($url->isExternal()) {
+                $link->addAttributes(['target' => '_blank']);
+            }
+
+            $urls[] = $link->render();
+
+            return '$objectLink' . (count($urls) - 1) . '$';
```

Comment:
Is there a reason why this doesn't return the transformed URL already?

Reply:
Yes, as you said the transformed url can be directly returned.